include/xnvme_be.h

```diff
@@ -77,6 +77,14 @@ struct xnvme_be_admin {
 	 */
 	int (*cmd_admin)(struct xnvme_cmd_ctx *, void *, size_t, void *, size_t);
 
+	/**
+	 * Pass pseudo admin-commands
+	 *
+	 * That is representation of various operations in NVMe commands, that look
+	 * like but are not actual NVMe commands.
+	 */
+	int (*cmd_pseudo)(struct xnvme_cmd_ctx *, void *, size_t, void *, size_t);
```

Comment:
This requires a flag to differentiate what the pseudo command needs to do (subsystem reset, or controller reset etc).

Reply:
The idea is that we would encode this in the NVMe-command. That is, in ``ctx->cmd``. E.g. somewhere, library-internal, we can keep a list of opcodes for the different operations. That is, we assign opcodes to the operations: (subsysystem-reset, controller-reset, namespace-rescan, show-reqs.). Then, in e.g. ``lib/xnvme_topology.c`` the functions would be implemented by in a form similar to command-construction for NVMe-command-sets.

Something like:

```c
  struct xnvme_cmd_ctx ctx = { 0 };
  int err;
  
  ctx.cmd.common.opcode = XNVME_SPEC_PSEUDO_OPC_CONTROLLER_RESET;

  err = xnvme_cmd_pass_pseudo(&ctx, NULL, 0, NULL, 0);

  return err | ctx.cpl.status.val;
```

Let me add those operations as pseudo-opcodes, and fill them out to better demonstrate this.

Reply:
Expanded with opcodes, xnvme_cmd_pass_pseudo(), and usage-examples.

Reply:
Yeah this should work